src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -10,46 +10,148 @@
+    llvm::Value* lhs = values.back();
+    values.pop_back();
+    llvm::Value* result;
+
+    // TODO: support more binary operators
+    switch(op) {
+#define DISPATCH(binary_op, llvm_op)                \
+    case binary_op:                                 \
+        result = llvm_op(lhs, rhs);                 \
+        values.push_back(result);                   \
+        break;
+
+        DISPATCH(ast::BinaryOp::BOP_ADDITION, builder.CreateFAdd);
+        DISPATCH(ast::BinaryOp::BOP_DIVISION, builder.CreateFDiv);
+        DISPATCH(ast::BinaryOp::BOP_MULTIPLICATION, builder.CreateFMul);
+        DISPATCH(ast::BinaryOp::BOP_SUBTRACTION, builder.CreateFSub);
+
+#undef DISPATCH
+    }
 }
 
-void CodegenLLVMVisitor::visit_procedure_block(const ast::ProcedureBlock& node) {
-    logger->info("CodegenLLVMVisitor : visiting {} procedure", node.get_node_name());
-
-    // print position, nmodl and json form as
-    /*
-    logger->info("Location {} \n NMODL {} \n JSON : {} \n",
-                 node.get_token()->position(),
-                 to_nmodl(node),
-                 to_json(node));
-    */
-    node.visit_children(*this);
-    // TODO : code generation for procedure block
+void CodegenLLVMVisitor::visit_boolean(const ast::Boolean &node) {
+    llvm::Value* constant = llvm::ConstantInt::get(llvm::Type::getInt1Ty(*context), node.get_value());
+    values.push_back(constant);
+}
+
+void CodegenLLVMVisitor::visit_double(const ast::Double &node) {
+    llvm::Value* constant = llvm::ConstantFP::get(llvm::Type::getDoubleTy(*context), node.get_value());
+    values.push_back(constant);
+}
+
+void CodegenLLVMVisitor::visit_integer(const ast::Integer &node) {
+    llvm::Value* constant = llvm::ConstantInt::get(llvm::Type::getInt32Ty(*context), node.get_value());
+    values.push_back(constant);
+}
+
+void CodegenLLVMVisitor::visit_local_list_statement(const ast::LocalListStatement &node) {
+    for (const auto& variable : node.get_variables()) {
+        // LocalVar always stores a Name
+        auto name = variable->get_node_name();
+        llvm::Type* var_type = llvm::Type::getDoubleTy(*context);
+        llvm::Value* alloca = builder.CreateAlloca(var_type, /*ArraySize=*/nullptr, name);
+        namedValues[name] = alloca;
+    }
 }
 
 void CodegenLLVMVisitor::visit_program(const ast::Program& node) {
     node.visit_children(*this);
-    result_code = "Hello World";
+    // Keep this for easier development (maybe move to debug mode later)
+    std::cout << print_module();
+}
+
+void CodegenLLVMVisitor::visit_procedure_block(const ast::ProcedureBlock& node) {
+    auto name = node.get_node_name();
+    const auto& parameters = node.get_parameters();
+
+    // The procedure parameters are doubles by default
+    std::vector<llvm::Type*> arg_types;
+    for (unsigned i = 0, e = parameters.size(); i < e; ++i)
+        arg_types.push_back(llvm::Type::getDoubleTy(*context));
+    llvm::Type* return_type = llvm::Type::getVoidTy(*context);
```

Comment:
nitpick : indentation

Reply:
Actually, what is the indentation used for the project? I noticed that headers use 2 spaces, and implementation uses 4 spaces. So I will just use that?

Reply:
Oh, I was mistaken here. I see that line 103 is indented for loop! 

For indentation, we use 4 spaces everywhere. In fact, assuming you have clang-format 8/9/10, and cmake option ` -DNMODL_CLANG_FORMAT=ON`,  doing `make clang-format` should format everything consistently!

If you see 2 spaces somewhere, let me know.